src/wallet/wallet.cpp

```diff
@@ -3964,6 +3965,27 @@ bool CWalletTx::IsImmatureCoinBase() const
     return GetBlocksToMaturity() > 0;
 }
 
+bool CWalletTx::isFinal() const
+{
+    return pwallet->CheckFinalWalletTx(*tx);
+}
+
+bool CWallet::CheckFinalWalletTx(const CTransaction& tx) const
+{
+    // Wallet not doing historical validation work, it's only
+    // interested by the nLocktime evaluation as enforced on
+    // current chain tip (BIP 113). In the future, if there
+    // is forks changing this, wallet should be updated.
+
+    // CheckFinalWalletTx() uses m_last_block_processed_height +1
+    // to evaluate nLockTime because when IsFinalTx() is called within
+    // CBlock::AcceptBlock(), the height of the block *being*
+    // evaluated is what is used. Thus if we want to know if a
+    // transaction can be part of the *next* block, we need to call
+    // IsFinalTx() with one more than m_last_block_processed_height.
+    return IsFinalTx(tx, m_last_block_processed_height + 1, m_last_block_median_time_past);
```

Comment:
In commit "Replace Chain::checkFinal by CWallet::CheckFinalWalletTx" (0ef43cef24b837367a90c97a31470e0f19f18552)

I see this comment is copied from [existing code](https://github.com/bitcoin/bitcoin/blob/6fef85bfa3cd7f76e83b8b57f9e4acd63eb664ec/src/validation.cpp#L200-L205), but the part about this function being called from `CBlock::AcceptBlock` doesn't apply here, right? Would it be ok to drop this long comment and simply write that we are passing  `m_last_block_processed_height + 1` because we are interested to know if the transaction could be included in the next block, not the current tip?

Reply:
The comment is about that we want to call `IsFinalTx` the same way it's called in `AcceptBlock`, i.e with `nBlockHeight` set to tip+1 to evaluate the consensus rule identically in both wallet and node. You don't want discrepancies between them. I think the last sentence is expressing that "if a transaction can be part of the *next* block", and speaking of `m_last_block_processed_height` is verbose but at least you avoid confusion between wallet tip and node tip.